### PR TITLE
Compose GalleryCommentsScreen

### PR DESCRIPTION
**Pull request by nullArrayList**

### CODE REVIEW

**Comment by nullArrayList** [comment]
Todo: 
Fab-TextField animation and predictive support
Edit existing comment
BBCode editor

But shouldn't block merging now, let's refine it later

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+                                mode = ListUrlBuilder.MODE_UPLOADER,
+                                mKeyword = item.user,
+                            )
+                            navigator.navAnimated(R.id.galleryListScene, lub.toStartArgs(), true)
+                        },
+                        onCardClick = {
+                            coroutineScope.launch {
+                                context.doCommentAction(item)
+                            }
+                        },
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:327]: Not working when clicking comment text

**Review by FooIbar — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt (hunk 1 above)
Comment (nullArrayList) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:327]: Yup, comment text has its own OnClickListener
I will try resolve it tomorrow

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+                            ) {
+                                Text(text = stringResource(id = R.string.click_more_comments))
+                            }
+                        }
+                    }
+                }
+            }
+            if (commenting) {
+                Surface(
+                    modifier = Modifier.align(Alignment.BottomCenter).imePadding(),
```
[the remark this answers is not in the document]
Comment (nullArrayList) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:375]: The surface should not have navbar padding
Let textfield take it

[the remark this answers is not in the document]
Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:375]: The send button as well

[the remark this answers is not in the document]
Comment (asuka-mio) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt]: Done

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+                FloatingActionButton(onClick = { commenting = true }) {
+                    Icon(imageVector = Icons.AutoMirrored.Default.Reply, contentDescription = null)
+                }
+            }
+        },
+    ) { paddingValues ->
+        val keylineMargin = dimensionResource(id = R.dimen.keyline_margin)
+        Box(modifier = Modifier.fillMaxSize()) {
+            LazyColumn(
+                modifier = Modifier.padding(horizontal = keylineMargin),
```
[the remark this answers is not in the document]
Comment (asuka-mio) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:276]: Done

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt (hunk 1 above)
Comment (asuka-mio) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:340]: Done

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt [hunk 4]
```diff
@@ -262,11 +265,20 @@ fun GalleryCommentsScreen(galleryDetail: GalleryDetail, navigator: NavController
         },
     ) { paddingValues ->
         val keylineMargin = dimensionResource(id = R.dimen.keyline_margin)
+        var editTextMeasured by remember { mutableStateOf(0.dp) }
         Box(modifier = Modifier.fillMaxSize().imePadding()) {
+            val additionalPadding = if (commenting) {
+                editTextMeasured
+            } else {
+                16.dp + 56.dp // Fab size + Fab space
+            }
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:274]: A bit too much when combined with the load more button

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+                    modifier = Modifier.align(Alignment.BottomCenter).onGloballyPositioned { coordinates ->
+                        editTextMeasured = with(density) { coordinates.size.height.toDp() }
+                    },
+                    color = MaterialTheme.colorScheme.primaryContainer,
+                ) {
+                    Row(modifier = Modifier.fillMaxWidth().navigationBarsPadding()) {
+                        BasicTextField2(
+                            value = userComment,
+                            onValueChange = { userComment = it },
+                            modifier = Modifier.weight(1f).padding(keylineMargin),
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:400]: The text size is too small compared with the send button

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt (hunk 4 above)
Comment (asuka-mio) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:277]: Done

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt (hunk 5 above)
Comment (asuka-mio) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt:396]: Done

**Review by FooIbar — APPROVED** [review]
LGTM.
Another thing is the comment list jumps on the first open of the edit panel, but I guess that could be resolved later.

**Pull request merged**